planning/autoware_path_optimizer/include/autoware/path_optimizer/node.hpp

```diff
@@ -75,6 +78,7 @@ class PathOptimizer : public rclcpp::Node
   bool enable_outside_drivable_area_stop_;
   bool enable_skip_optimization_;
   bool enable_reset_prev_optimization_;
+  bool mrm_required_{false};
```

Comment:
Publishing the diagnostics does not mean triggering the MRM since it depends on the setting of the system packages.
So, `is_optimization_failed_` or something would be proper.